src/runtime/composables.ts

```diff
@@ -1,15 +1,17 @@
+import type { H3Event } from 'h3'
 import type { $Img } from '../module'
 
 import { createImage } from './image'
 import { imageOptions } from '#build/image-options.mjs'
 import { useNuxtApp, useRuntimeConfig } from '#imports'
 
-export const useImage = (): $Img => {
+export const useImage = (event?: H3Event): $Img => {
   const config = useRuntimeConfig()
   const nuxtApp = useNuxtApp()
```

Comment:
We are emulating `nuxtApp` for server no? Would be nice to typecase it to limited type (only `baseURL` is available rest are pertial) to avoid future assumptions

Reply:
yes, we already do this:

https://github.com/nuxt/image/blob/9a5e4965c826997658215b298f80575ab96255b2/src/types/image.ts#L40-L42

we don't use nuxtApp in the server (we only call `createImage` for nitro endpoints) - nuxtApp is only used in the vue version of `useImage`